syntax = "proto3";

package v1;

import "google/protobuf/timestamp.proto";

option go_package = "github.com/pachirode/ffmpeg-study/pkg/api/apiserver/v1;v1";

// 音频流元信息
message AudioMeta {
    // codec 编解码器，例如 aac、mp3
    string codec = 1;
    // sample_rate 采样率
    int32 sample_rate = 2;
    // channels 通道数
    int32 channels = 3;
    // pts 当前帧时间戳
    google.protobuf.Timestamp pts = 4;
}

// 音频数据包
message AudioPacket {
    // meta 音频帧元信息
    AudioMeta meta = 1;
    // data 编码后的音频帧数据
    bytes data = 2;
    // createdAt 当前帧创建时间
    google.protobuf.Timestamp createdAt = 3;
}

// 请求开始音频流
message StartAudioStreamRequest {
    // stream_id 唯一流ID
    string stream_id = 1;
}

// 响应
message AudioStreamResponse {
    // success 是否成功
    bool success = 1;
    // message 描述信息
    string message = 2;
}

